thrust/device_vector.h

```diff
@@ -100,6 +103,31 @@ template<typename T, typename Alloc = thrust::device_malloc_allocator<T> >
     device_vector(const device_vector &v)
       :Parent(v) {}
 
+  #if __cplusplus >= 201103L
+    /*! Move constructor use the move semantic over an exemplar
+     * vector_base.
+     *  \param v The vector_base to copy.
+     */
+     __host__
+    device_vector(device_vector &&v)
+      :Parent(std::forward<Parent>(v)) {}
+
+    /*! Move assign operator use the move semantic over an exemplar
+     * vector_base.
+     *  \param v The vector_base to copy.
+     */
+     __host__
+     device_vector &operator=(device_vector &&v)
+     { Parent::operator=(std::forward<Parent>(v)); return *this; }
+
+     /*! Assign operator copies from an exemplar \p device_vector with different type.
+      *  \param v The \p device_vector to copy.
+      */
+     __host__
+     device_vector &operator=(const device_vector &v)
```

Comment:
To make it short, my intend was to make the least possible changes to the non-C++11 behaviour/dependencies of the library.
As said in the initial pull request post, the declaration of the move assignment made the implicit copy assignment deleted, doing so, I first had to face a compile time error, complaining about it.
So I had to declare it explicitly, at least for __host__, to avoid error. But as the implicitly declared one works fine in the non-c++11 compilation, I did not wanted to take the risk of changing the previous behaviour in this case.